ui/raidboss/data/05-shb/trial/ruby-weapon-ex.js

```diff
@@ -2,10 +2,130 @@
 
 [{
   zoneRegex: {
-    en: /^Cinder Drift \(Extereme\)$/,
+    en: /^Cinder Drift \(Extreme\)$/,
   },
   timelineFile: 'ruby-weapon-ex.txt',
   triggers: [
+    {
+      id: 'RubyEx Optimized Ultima',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4ABE', capture: false }),
+      condition: function(data) {
+        return data.role == 'healer' || data.role == 'tank' || data.CanAddle();
+      },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'RubyEx Stamp',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4B03' }),
+      condition: function(data) {
+        return data.role == 'tank' || data.role == 'healer';
+      },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'RubyEx Undermine',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4AD0', capture: false }),
+      preRun: function(data) {
+        data.rubyCounter = data.rubyCounter || 0;
+        data.rubyCounter++;
+      },
+      alertText: {
+        en: 'Away From Lines',
+      },
+    },
+    {
+      id: 'RubyEx Liquefaction',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4AEC', capture: false }),
+      preRun: function(data) {
+        data.rubyCounter = data.rubyCounter || 0;
+        data.rubyCounter++;
+      },
+      condition: function(data) {
+        return data.rubyCounter % 2;
+      },
+      alertText: {
+        en: 'Get On Lines',
+      },
+    },
+    {
+      id: 'RubyEx Ruby Ray',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4B02', capture: false }),
+      alertText: {
+        en: 'Away From Front',
+      },
+    },
+    {
+      id: 'RubyEx Cut And Run',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4B05', capture: false }),
+      infoText: {
+        en: 'Go Sides',
+      },
+    },
+    {
+      id: 'RubyEx High-Powered Homing Lasers',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4AD8' }),
+      condition: Conditions.targetIsYou(),
+      infoText: {
+        en: 'Stack on YOU',
+      },
+    },
+    {
+      id: 'RubyEx Optimized Ultima',
+      regex: Regexes.startsUsing({ source: 'The Ruby Weapon', id: '4B2D', capture: false }),
+      alertText: {
+        en: 'Enrage!',
```

Comment:
Maybe timeline only?

Reply:
Only concern is that enrage can happen mid-rotation -- like if you push the boss to ~5% it will start enraging, even if there's still mechanics left in the timeline

Reply:
No, that's fair.  Can you leave a comment and say that this fight can have a ridiculously early enrage sometimes, so it needs a warning?  I don't want other people seeing this and making enrage triggers for everything.